otherlibs/unix/unix.mli

```diff
@@ -675,11 +675,39 @@ val close_process_full :
 (** {6 Symbolic links} *)
 
 
-val symlink : string -> string -> unit
-(** [symlink source dest] creates the file [dest] as a symbolic link
-   to the file [source].
-
-   On Windows: not implemented. *)
+val symlink : ?is_dir:bool -> string -> string -> unit
+(** [symlink ?is_dir source dest] creates the file [dest] as a symbolic link
+   to the file [source]. On Windows, [~is_dir] indicates if the symbolic link
+   points to a directory or a file; if omitted, [symlink] examines [source]
+   using [stat] and picks appropriately, if [source] does not exist then [false]
+   is assumed. On Unix, [~is_dir] ignored.
```

Comment:
If it were a Windows-only implementation, I'd go for `is_dir` being mandatory - but the potentially slow default is also the least painful cross-platform solution. By having the `is_dir` unspecified case attempt to work out the correct one using `stat`, you get the closest possible behaviour to a Posix symlink on Windows.
I'm entirely easy on the name of the parameter - you came up with `is_dir` :wink: